### ISSUE TITLE
DockerCompose.ps() incorrectly ignores certain valid container names

### ISSUE THREAD

**Issue by j-baker** [palantir/docker-compose-rule#38]
To parse results, it calls https://github.com/palantir/docker-compose-rule/blob/505286255bee81cec0f537e1cb7a09aafb21e570/src/main/java/com/palantir/docker/compose/connection/ContainerNames.java#L48

which assumes that the container name is in the format PROJECT_name_SOMENUMBER, I think.

But, you can set the container name to whatever you want, and the name can also have underscores in.

In these cases, the container name will be filtered out.

For example, if the result of docker-compose ps is

```
47aab4ab_salt_config_1       /true                            Exit 0                                                              
cassandra.palantir.dev       /docker-entrypoint.sh cass ...   Up       7000/tcp, 7001/tcp, 7199/tcp, 9042/tcp, 9160/tcp           
```

then no containers will be returned.

**Comment by joelea** [comment]
Loving the name change on the issue :+1: Though I do like the old name better.

This sucks and doesn't have an easy fix, I have a few ideas I'm going to let stew though. If you have any ideas, feel free to mention them or make a PR.

**Comment by felixdesouza** [comment]
I was looking this up at some point. I think terminology needs to be cleared up also. 

You can list the the services (what you specify in your docker-compose file, regardless of `container_name` and also project name) that your compose file provides. If you're not doing any sort of scaling then there is a 1-1 mapping of docker-compose services (as defined by docker) and a container that runs an instance of that service. You can do this with:
```
docker-compose config --services
```
This requires docker-compose version 1.6 and I guess transitively docker 1.10 (although not entirely sure)

Not entirely sure what the scope is for supporting scaling especially within JUnit code, but suppose we didn't support scaling via `docker-compose scale service1=<num-s1> service2=<num-s2> ...` then we could do something like:

```
for service in 'docker-compose config --services':
    psOutput = docker-compose ps service
    parse container name(psOutput)
```

Regarding name clarification, I think our definition of a service may or may not be confusing given how docker-compose has its own concept.

As an aside, I was also thinking, as opposed to having to do a lot of parsing and regexing ourselves, once we know which containers are attached to which service, we could just connect to the docker API, which is just a HTTP endpoint. This would give us anything that we need to parse to infer state such as port mappings. It's not directly related, but fixing the issue in the way above allows this to be possible.

**Comment by joelea** [comment]
oooh, docker-compose config --services is nice. That looks like the right kind of path to go down. Thanks1

**Comment by felixdesouza** [comment]
As an extra case to this, if you have a dash in your service name e.g. `d-c-r`, it will drop `-c-r`. I looked briefly and it looks like this is the culprit:
https://github.com/palantir/docker-compose-rule/blob/master/src/main/java/com/palantir/docker/compose/connection/ContainerNames.java#L43 
```java
public static ContainerNames parseFromDockerComposePs(String psOutput) {
    String[] splitOnSeparator = psOutput.split("-+\n"); // <= this guy
    if (splitOnSeparator.length < 2) {
        return new ContainerNames(emptyList());
    }
    return new ContainerNames(getContainerNamesAtStartOfLines(splitOnSeparator[1]));
}
```

**Comment by iamdanfox** [comment]
I think the PR @CRogers just merged (#53) might help with this issue - we can now guarantee that a running container will either be named: `project_service_x` or random: `ac34bc56_service_x`

**Comment by iamdanfox** [comment]
Fixed in the above PR (https://github.com/palantir/docker-compose-rule/pull/127) see "Use docker-compose config --services instead of ps"
